core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -140,6 +140,81 @@ export class AdminBackendApiService {
       });
     });
   }
+
+  viewUsersRole(
+      filterCriterion:string, role:string, username:string
+  ):Promise<ArrayBuffer> {
+    return this.http.get<ArrayBuffer>(
+      AdminPageConstants.ADMIN_ROLE_HANDLER_URL, {
+        params: {
+          filter_criterion: filterCriterion,
+          role: role,
+          username: username
+        }
+      }
+    ).toPromise();
+  }
+
+  updateUserRole(
+      newRole:string, username:string, topicId:string
+  ):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_ROLE_HANDLER_URL, {
+        role: newRole,
+        username: username,
+        topic_id: topicId
+      }
+    ).toPromise();
+  }
+
+  addContributionReviewer(
+      category:string, username:string, languageCode:string
+  ):Promise<void> {
+    return this.http.post<void>(
+      AdminPageConstants.ADMIN_JOB_ADD_CONTRIBUTION_REVIEWER_HANDLER, {
+        review_category: category,
+        username: username,
+        language_code: languageCode
+      }
+    ).toPromise();
+  }
+
+  viewContributionReviewers(
+      category:string, languageCode:string,
+  ):Promise<ArrayBuffer> {
+    return this.http.get<ArrayBuffer>(
```

Comment:
Why the type is ArrayBuffer? I was expecting the response to be a dict

[Please specify the correct type here and elsewhere!)

Reply:
done!